diff --git BUILD BUILD
new file mode 100644
index 0000000..65c81af
--- /dev/null
+++ BUILD
@@ -0,0 +1,19 @@
+cc_library(
+    name = "aws-c-cal",
+    srcs = glob([
+        "source/*.c",
+        "source/unix/*.c",
+    ]),
+    hdrs = glob([
+        "include/aws/cal/*.h",
+        "include/aws/cal/private/*.h",
+    ]),
+    includes = [
+        "include",
+    ],
+    deps = [
+        "@aws-c-common",
+        "@boringssl//:ssl",
+    ],
+    visibility = ["//visibility:public"],
+)
diff --git bin/BUILD bin/BUILD
new file mode 100644
index 0000000..b05b316
--- /dev/null
+++ bin/BUILD
@@ -0,0 +1,17 @@
+cc_binary(
+    name = "produce_x_platform_fuzz_corpus",
+    srcs = ["produce_x_plactform_fuzz_corpus/main.c"],
+    deps = ["//:aws-c-cal"],
+)
+
+cc_binary(
+    name = "run_x_platform_fuzz_corpus",
+    srcs = ["run_x_platform_fuzz_corpus/main.c"],
+    deps = ["//:aws-c-cal"],
+)
+
+cc_binary(
+    name = "sha256_profile",
+    srcs = ["sha256_profile/main.c"],
+    deps = ["//:aws-c-cal"],
+)
