diff --git BUILD BUILD
new file mode 100644
index 0000000..9609392
--- /dev/null
+++ BUILD
@@ -0,0 +1,37 @@
+cc_library(
+    name = "aws-crt-cpp",
+    srcs = glob([
+        "source/*.cpp",
+        "source/auth/*.cpp",
+        "source/cbor/*.cpp",
+        "source/crypto/*.cpp",
+        "source/endpoint/*.cpp",
+        "source/http/*.cpp",
+        "source/io/*.cpp",
+        "source/iot/*.cpp",
+        "source/mqtt/*.cpp",
+        "include/aws/crt/mqtt/private/*.h",
+    ]),
+    hdrs = glob([
+        "include/aws/crt/*.h",
+        "include/aws/crt/auth/*.h",
+        "include/aws/crt/cbor/*.h",
+        "include/aws/crt/crypto/*.h",
+        "include/aws/crt/endpoints/*.h",
+        "include/aws/crt/http/*.h",
+        "include/aws/crt/io/*.h",
+        "include/aws/crt/mqtt/*.h",
+        "include/aws/iot/*.h",
+    ]),
+    includes = [
+        "include",
+    ],
+    deps = [
+        "@aws-c-auth",
+        "@aws-c-event-stream",
+        "@aws-c-io",
+        "@aws-c-mqtt",
+        "@aws-c-s3",
+    ],
+    visibility = ["//visibility:public"],
+)
diff --git include/aws/crt/Config.h include/aws/crt/Config.h
new file mode 100644
index 0000000..9b4f5dd
--- /dev/null
+++ include/aws/crt/Config.h
@@ -0,0 +1,11 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#define AWS_CRT_CPP_VERSION "0.27.3"
+#define AWS_CRT_CPP_VERSION_MAJOR 0
+#define AWS_CRT_CPP_VERSION_MINOR 27
+#define AWS_CRT_CPP_VERSION_PATCH 3
+#define AWS_CRT_CPP_GIT_HASH "17e30acb7d6338a7b180c0cd522de0b4c51ceadd"
